plugins/outputs/nats/nats.go

```diff
@@ -19,20 +25,68 @@ import (
 var sampleConfig string
 
 type NATS struct {
-	Servers     []string      `toml:"servers"`
-	Secure      bool          `toml:"secure"`
-	Name        string        `toml:"name"`
-	Username    config.Secret `toml:"username"`
-	Password    config.Secret `toml:"password"`
-	Credentials string        `toml:"credentials"`
-	Subject     string        `toml:"subject"`
-
+	Servers     []string                `toml:"servers"`
+	Secure      bool                    `toml:"secure"`
+	Name        string                  `toml:"name"`
+	Username    config.Secret           `toml:"username"`
+	Password    config.Secret           `toml:"password"`
+	Credentials string                  `toml:"credentials"`
+	Subject     string                  `toml:"subject"`
+	Jetstream   *JetstreamConfigWrapper `toml:"jetstream"`
 	tls.ClientConfig
 
 	Log telegraf.Logger `toml:"-"`
 
-	conn       *nats.Conn
-	serializer serializers.Serializer
+	conn            *nats.Conn
+	jetstreamClient jetstream.JetStream
+	serializer      serializers.Serializer
+}
+
+type JetstreamConfigWrapper struct {
+	jetstream.StreamConfig
+}
+
+func (jw *JetstreamConfigWrapper) UnmarshalTOML(data []byte) error {
```

Comment:
After chatting, we would prefer you not unmarhsall the TOML by hand here. Rather you can either verify that the TOML is parsed correctly based on the import or create an explicate struct with TOML tags. 

After that we should be good to go with this PR!

Reply:
Creating explicit struct would be a maintenance overhead if the config params change either in data types or new field additions. I would personally avoid it.

Not sure how to verify the correctness of the TOML. Can you give some thoughts?

Reply:
We would prefer the explicit struct. Yes, if upstream changes things we would need to adapt, but parsing config TOML in a plugin is far from ideal.

Adding the TOML tags should be all that you need.

Reply:
I have made the required changes. Thanks!